src/tests/conformance/messages.rs

```diff
@@ -564,67 +564,121 @@ async fn eagerly_crawls_network_for_peers() {
     node.stop().await;
 }
 
-// #[tokio::test]
+#[tokio::test]
 async fn correctly_lists_peers() {
     // ZG-CONFORMANCE-013
     //
     // The node responds to a `GetAddr` with a list of peers it’s connected to. This command
     // should only be sent once, and by the node initiating the connection.
     //
+    // In addition, this test case exercises the known zebra bug: https://github.com/ZcashFoundation/zebra/pull/2120
+    //
     // Test procedure
-    //  Start a node, and sequentially for each peer `i` of `N`:
-    //      1. Initiate a connection and complete the handshake
-    //      2. Send `GetAddr` request
-    //      3. Receive `Addr` response
-    //      4. Verify `Addr` contains list of previous `i-1` peers
+    //      1. Establish N peer listeners
+    //      2. Start node which connects to these N peers
+    //      3. Create i..M new connections which,
+    //          a) Connect to the node
+    //          b) Query GetAddr
+    //          c) Receive Addr == N peer addresses
     //
-    // This test currently fails for zcashd and zebra passes.
+    // This test currently fails for both zcashd and zebra.
     //
     // Current behaviour:
     //
     //  zcashd: Never responds. Logs indicate `Unknown command "getaddr" from peer=1` if we initiate
     //          the connection. If the node initiates the connection then the command is recoginized,
     //          but likely ignored (because only the initiating node is supposed to send it).
     //
-    //  zebra:  Infinitely spams `GetAddr` and `GetData`. Can be coaxed into responding correctly if
-    //          all its peer connections have responded to `GetAddr` with a non-empty list.
+    //  zebra:  Times out due to infinitely spamming `GetAddr` and `GetData`.
+    //
+    //          Can be coaxed into responding by sending a non-empty Addr in
+    //          response to node's GetAddr. This still fails as it includes previous inbound
+    //          connections in its address book (as in the bug listed above).
```

Comment:
Have you tried running with latest `main` or after the bug was supposedly fixed?

Reply:
I wasn't aware it was in master. I've re-run the test and updated the result.

It now no longer spams, but it remains in "network setup" state until we respond to the `GetAddr` in which case it then responds.